### PR TITLE
Add Hexagon layer methods

### PR DESCRIPTION
**Pull request by heshan0131**
- Add `getHexagons` and `getSortedCounts` methods

### CODE REVIEW

**Comment by heshan0131** [comment]
@shaojingli Why are you approving pull requests, aren't you suppose to be on vacation?

**Comment by Pessimistress** [comment]
Update layer docs?

**Comment by heshan0131** [comment]
This change doesn't affect the API. `getHexagons` and `getSortedCounts` are methods of the layer class. separate them out will make it easier to create layer subclasses and override them.

File: src/layers/core/hexagon-layer/hexagon-layer.js [hunk 1]
```diff
@@ -125,6 +121,18 @@ export default class HexagonLayer extends Layer {
     }
   }
 
+  getHexagons() {
+    const {hexagonAggregator} = this.props;
+    const {viewport} = this.context;
+    const hexagons = hexagonAggregator(this.props, viewport);
+    Object.assign(this.state, {hexagons});
```
Comment (ibgreen) [inline src/layers/core/hexagon-layer/hexagon-layer.js]: Nit: we have a `Layer.setState` method that also sets the dirty flag to ensure re-render.

**Pull request merged**